src/kata6.joinNames.js

```diff
@@ -1,3 +1,10 @@
-const joinNames = namesObj => {};
+const joinNames = namesObj => {
+  const stringOfNames = namesObj.reduce((accumulator, person, index) => {
+    if (index !== namesObj.length - 1) return `${accumulator}, ${person.name}`;
```

Comment:
An alternative here is you could also check for index of 0 and just return person.name so you don't have to do the slice later.

Reply:
Thanks for this, I was so confused, I was trying to do this but using `accumulator.name` for index 0, without the default value but I was getting [object Object] and I couldn't figure out why so for example:

```javascript
const stringOfNames = namesObj.reduce((accumulator, person, index) => {
    if (index === 0) return `${accumulator.name}`;
    if (index !== namesObj.length - 1) return `${accumulator}, ${person.name}`;
    return `${accumulator} & ${person.name}`;
  });
  ```
  
  I still don't fully understand why I don't think, because shouldn't accumulator.name be accessing the first object in the arrays name property here?
  
  Anyway, your solution worked but I would like to know why my attempt was returning [object Object] if you could explain?
  
  This is the updated solution the way you said, thanks it's definitely cleaner than the slice on the end it was driving me crazy
  ```javascript
  const stringOfNames = namesObj.reduce((accumulator, person, index) => {
    if (index === 0) return `${person.name}`;
    if (index !== namesObj.length - 1) return `${accumulator}, ${person.name}`;
    return `${accumulator} & ${person.name}`;
  }, "");
  ```

Reply:
So it's because the default accumulator if you don't pass a second argument to reduce is the first element of the array. The array is an array of person objects, therefore the first accumulator is an object and then we are string interpolating it in the `return `${accumulator} & ${person.name}`;` So you get a string representation of an object. The second solution here is initialising the accumulator as an empty string.

Reply:
Oh, I think I understand now... On my first go, I was expecting accumulator.name on index 0 to be bart (with no default accumulator value) but I have just realised it starts from index 1 when you don't pass the default accumulator value which is kind of weird, if I did my first example but looked for index 1 it works, but either way, I like yours better. Something I'll have to keep in mind is that the index changes depending on if you pass a default accumulator parameter or not! Thanks for the help :)

Reply:
So the accumulator takes the entire first index if you don't pass a default for it. So you won't get person.name, you'll get the whole person object, the only way you could specify that you wanted the name key is if you had passed it as a default/initialValue for the accumulator as the second argument of the reduce. But yes, if you look at the reduce documentation on mdn, it shows you that the first argument of the callback function that you pass into a reduce, previousValue, is set to initialValue (if you set one) or array[0] (if you didn't) and then the second param of your callback function is set to array[0] (if you set an initialValue) and array[1] (if you didn't). Reduce is a little complicated to learn but once you've practiced with it, it starts to make sense. Most of the time when I am working with reduce in production code, I am using complex data types so it is easiest to set a default value for the accumulator.